src/site/content/en/blog/window-controls-overlay/index.md

```diff
@@ -0,0 +1,432 @@
+element draggable.
+
+```css
+/* The entire search `div` is draggable… */
+.search {
+  app-region: drag;
+}
+
+/* …except for the `input`. */
+input {
+  app-region: no-drag;
+}
+```
+
+With this CSS in place, the user can drag the app window as usual by dragging the `div`, the `img`,
+or the `label`. Only the `input` element is interactive so the search query can be entered.
+
+### Feature detection
+
+Support for Window Controls Overlay can be detected by testing for the existence of
+`windowControlsOverlay`:
+
+```js
+if ('windowControlsOverlay' in navigator) {
+  // Window Controls Overlay is supported.
+}
+```
+
+### Querying the window controls region with `windowControlsOverlay`
+
+The code so far has only one problem: on some platforms the window controls are on the right, on
+others they are on the left. This means that the linear gradient background image needs to be
+dynamically adapted to run from `#131313` to `maroon` or vice versa, so that it blends in with the
+title bar's `maroon` background color that is determined by
+`<meta name="theme-color" content="maroon">`. This can be achieved by querying the
+[`getBoundingClientRect()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect)
+API on the `navigator.windowControlsOverlay` property.
+
+```js
+if ('windowControlsOverlay' in navigator) {
+  const { x } = navigator.windowControlsOverlay.getBoundingClientRect();
+  // Window controls are on the right (like on Windows).
+  // [ windowControlsOverlay___________________ [_] [■] [X] ]
+  if (x === 0) {
+    div.classList.add('search-controls-right');
+  }
+  // Window controls are on the left (like on macOS).
+  // [ [X] [_] [■] ___________________windowControlsOverlay ]
+  else {
+    div.classList.add('search-controls-left');
+  }
+} else {
+  // When running in a non-supporting browser tab.
+  div.classList.add('search-controls-right');
+}
+```
+
+Rather than having the background image in the `.search` class CSS rules directly (as before), the
+modified code now uses two classes that the code above sets dynamically.
+
+```css
+.search-controls-left {
+  background-image: linear-gradient(90deg, maroon, 66%, #131313);
+}
+
+.search-controls-right {
+  background-image: linear-gradient(90deg, #131313, 33%, maroon);
+}
+```
+
+### Determining if the window controls overlay is visible
+
+The window controls overlay will not be visible in the title bar area in all circumstances. While it
+will naturally not be there on browsers that do not support the Window Controls Overlay feature, it
+will also not be there when the PWA in question runs in a tab. To detect this situation, you can
+query the `visible` property of the `windowControlsOverlay`:
+
+```js
+if (navigator.windowControlsOverlay.visible) {
+  // The window controls overlay is visible in the title bar area.
+}
+```
+
+{% Aside %} The window controls overlay visibility is not to be confused with visibility in the CSS
```

Comment:
The way I read this:
> Even if you set `display: none` on the window controls overlay, the `visible` property would
still report `true`.

It seems like it seems to suggest that a dev could modify the window controls overlay, but the overlay is a piece of browser UI that cannot be modified by the webcontent. By "window controls overlay" are you referring to the pseudo-titlebar div that's next to the overlay?

Reply:
I'm referring to… 

```html
<div class="search">
  <img src="chuck-norris.png" alt="Chuck Norris" width="32" height="32" />
  <label>
    <input type="search" />
    Search words in jokes
  </label>
</div>
```

…in my example. I need to phrase it like the `div` with the search bar that gets displayed in the window controls overlay.